docs/src/pages/components/data-grid/rendering/AntDesignGrid.js

```diff
@@ -123,14 +123,280 @@ CustomPagination.propTypes = {
     current: PropTypes.object.isRequired,
   }).isRequired,
   /**
-   * The object containing all pagination details in [[PaginationState]].
+   * The GridState object containing the grid current state.
    */
-  pagination: PropTypes.shape({
-    page: PropTypes.number.isRequired,
-    pageCount: PropTypes.number.isRequired,
-    pageSize: PropTypes.number.isRequired,
-    paginationMode: PropTypes.oneOf(['client', 'server']).isRequired,
-    rowCount: PropTypes.number.isRequired,
+  state: PropTypes.shape({
+    columnMenu: PropTypes.shape({
+      field: PropTypes.string,
+      open: PropTypes.bool.isRequired,
+    }).isRequired,
+    columnReorder: PropTypes.shape({
+      dragCol: PropTypes.string.isRequired,
+    }).isRequired,
+    columns: PropTypes.shape({
+      all: PropTypes.arrayOf(PropTypes.string).isRequired,
+      lookup: PropTypes.object.isRequired,
+    }).isRequired,
```

Comment:
@eps1lon Is there a way to customize the prop-types generation logic so it doesn't get into such a fine verbose fine-grain detail?

Reply:
`ttp.parseFromProgram` has an option for that called `shouldResolveObject`. We currently use this to reduce the verbosity of `classes` and `theme`: 

https://github.com/mui-org/material-ui/blob/6b00694698015633558a713cc68b9e841380d9be/scripts/generateProptypes.ts#L206-L211

Does this help here?

Reply:
@eps1lon Oh yeah, great idea, this works:

```diff
diff --git a/docs/scripts/formattedTSDemos.js b/docs/scripts/formattedTSDemos.js
index cdeff4af..d8b28355 100644
--- a/docs/scripts/formattedTSDemos.js
+++ b/docs/scripts/formattedTSDemos.js
@@ -85,7 +85,7 @@ async function transpileFile(tsxPath, program, ignoreCache = false) {

     const propTypesAST = typescriptToProptypes.parseFromProgram(tsxPath, program, {
       shouldResolveObject: ({ name }) => {
-        if (name === 'classes') {
+        if (name === 'classes' || name === 'state') {
           return false;
         }
```